src/System.Threading.Tasks.Channels/System/IO/Channels/ChannelOptions.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO.Channels
+{
+    /// <summary>Provides options that control the behavior of channel instances.</summary>
+    public abstract class ChannelOptions
```

Comment:
Is there a reason for the hierarchy? Would it be better to have no hierarchy and make the options types structs?

Reply:
I don't have a strong opinion.  I thought this is what we'd decided on, but I'm fine changing it if that's preferred.